eng/nativepgo.targets

```diff
@@ -0,0 +1,56 @@
+    <NativeOptimizationDataSupported Condition="'$(NoPgoOptimize)' == 'true'">false</NativeOptimizationDataSupported>
+    <NativeOptimizationDataSupported Condition="'$(DotNetBuildFromSource)' == 'true'">false</NativeOptimizationDataSupported>
+
+    <_NativeOptimizationDataPackageTarget>$(TargetOS.ToLower())-$(TargetArchitecture.ToLower())</_NativeOptimizationDataPackageTarget>
+    <_NativeOptimizationDataPackageTarget Condition="'$(TargetOS)' == 'windows'">windows_nt-$(TargetArchitecture.ToLower())</_NativeOptimizationDataPackageTarget>
+
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="optimization.$(_NativeOptimizationDataPackageTarget).PGO.CoreCLR"
+      Version="$(optimizationPGOCoreCLRVersion)"
+      Condition="'$(optimizationPGOCoreCLRVersion)'!='' And '$(NativeOptimizationDataSupported)'!=''"
+      GeneratePathProperty="true" />
+  </ItemGroup>
+
+
+  <!--                                                                       -->
+  <!-- Task: GetPgoDataPackagePath                                          -->
+  <!--                                                                       -->
+  <!-- Notes:                                                                -->
+  <!--                                                                       -->
+  <!-- DumpPgoDataPackagePath is used to get the path of the native PGO data -->
+  <!-- for other MSBuild projects, generally to pass to another project or   -->
+  <!-- native script like build-runtime.cmd/sh.                              -->
+  <!--                                                                       -->
+
+  <Target Name="GetPgoDataPackagePath" Returns="$(PgoPackagePath)">
+    <PropertyGroup>
+      <PgoPackagePathProperty>Pkgoptimization_$(_NativeOptimizationDataPackageTarget)_PGO_CoreCLR</PgoPackagePathProperty>
+    </PropertyGroup>
+
+    <!--
+      Use an item group for expansion of $($(PgoPackagePathProperty)) (an illegal MSBuild expression)
+      i.e. the prop value's value.
+    -->
+    <ItemGroup>
+      <PgoPackagePathPropertyItemList Include="$(PgoPackagePathProperty)" />
+      <PgoPackagePathPropertyItemList>
+        <PgoPackagePath>$(%(Identity))</PgoPackagePath>
+      </PgoPackagePathPropertyItemList>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <PgoPackagePath>@(PgoPackagePathPropertyItemList->'%(PgoPackagePath)')</PgoPackagePath>
+    </PropertyGroup>
+
+    <Error Condition="!Exists('$(PgoPackagePath)') And '$(NativeOptimizationDataSupported)' == 'true'" Text="Unable to locate restored PGO package at $(PgoPackagePath). Maybe the platform-specific package naming changed?" />
+  </Target>
+
+  <Target Name="OutputPgoPathForCI" Condition="'$(ContinuousIntegrationBuild)' == 'true' and '$(NativeOptimizationDataSupported)' == 'true'" DependsOnTargets="GetPgoDataPackagePath">
+    <Message Text="##vso[task.setvariable variable=CoreClrPgoDataArg]-pgodatapath &quot;$(PgoPackagePath)&quot;" />
```

Comment:
I am not a fan of the simple build having this communication mechanism bound to the CI provider. Before you could do build tools and managed components once and then loop over the native several times. Now it differs a little bit as you'll likely not copy the output of PgoPackagePath into the native build command line.

Reply:
Basically - was the txt file at a known location working so poorly?

Reply:
The "txt file at a known location" was all internal to the build-runtime script, so it wasn't really a well-known location.

Also, since the new design passes the PGO location as a parameter to the script, I wanted to make CI consistent and have it do the same thing that the `clr.native` subset does. Also, using the known script location would require some mechanism to mark the location as out-of-date if someone wanted to do a non-PGO build, whereas when the path is passed into the script, you can just opt out by not passing the path into the script.

Reply:
The problem is now the `build clr` and `build nativeprereqs+corelib`  with `build-runtime` won't work - which is precisely what people do to avoid the msbuild overhead. And the `vso` part is sad, but I can't think anything better.

Reply:
That workflow will work. The developer will just have to pass in a path to the PGO data path to `build-runtime`, which is generally unneeded for local workflows (but was nice to have as a default).

I don't believe many people use `build-runtime` directly nowadays outside of CI scenarios with all the improvements we've done to the root build script infra.

Reply:
How many of those people build in release mode and need the PGO optimization for their local build?

If they're trying to avoid the startup cost of MSBuild, they're probably already passing `-skipgenerateversion` and `-skiprestoreoptdata` to avoid the cost of MSBuild, so there'd be no change in the versioning workflow and minimal change in the PGO workflow.

If they don't need PGO optimization, then their current workflow will continue to work.

Reply:
You know - I considered all possibilities and I'm convinced it won't matter. If they want to repro a CI issue they can copy the commands from the build and adjust. Otherwise this is good enough for inner loop. The reliance on VSO is the only thing that makes the knee jerk happen, but we can always revisit that as needed - it's a tiny thing.